actions/deploy-renku/deploy-dev-renku.py

```diff
@@ -210,30 +198,51 @@ def configure_requirements(tempdir, reqs, component_versions):
     pprint.pp(reqs)
 
     helm_command = [
-            "helm",
-            "upgrade",
-            "--install",
-            release,
-            "./renku",
-            "-f",
-            values_file,
-            "--namespace",
-            namespace,
-            "--timeout",
-            "20m",
-        ]
+        "helm",
+        "upgrade",
+        "--install",
+        release,
+        "./renku",
+        "-f",
+        values_file,
+        "--namespace",
+        namespace,
+        "--timeout",
+        "20m",
+    ]
+
+    additional_values = {}
 
     if args.anonymous_sessions:
-        helm_command += ["--set", "global.anonymousSessions.enabled=true"]
+        additional_values = {**additional_values, **{"global": {"anonymousSessions": {"enabled": True}}}}
 
     if args.tests_enabled:
-        helm_command += [
-            "--set", "tests.enabled=true",
-            "--set", "tests.parameters.username=\"renku-test\"",
-            "--set", "tests.parameters.fullname=\"Renku Bot\"",
-            "--set", "tests.parameters.email=\"renku@datascience.ch\"",
-            "--set", "tests.parameters.password=\"$RENKU_BOT_DEV_PASSWORD\""
-        ]
+        additional_values = {
```

Comment:
It's already useful outside a github action, I use it all the time :) 

Here I agree though that there should be a nicer mechanism for adding additional values. Probably out of scope for this PR. However, the specific values for our tests on dev should not be hard-coded in this script but instead provided via the values file, i.e. I would simply remove the whole part about following

```
if args.tests_enabled:
```

Reply:
I think we need _some_ mechanism already within the scope of this PR to provide values which are specific to each ci deployment and can thus not be provided through the values file. The other option would be to modify the values file before passing it to the script.

Reply:
Yes, all the static values have been migrated to `dev-ci-values`, I will soon refactor this part of code where only the ENV for the secret will be loaded.
On the other hand if we want to handle the secrets in the `ops-dev.renku.ch` repository as well (through SOPS as usual) then yes we can entirely remove this section about `additional_values`. Should we go for this last option?

Reply:
That last option would be the cleanest imho.

Reply:
@ableuler which values need to be passed separately for each ci deployment?

Reply:
Strictly speaking it's only the `GITHUB_SHA` at the moment, everything else (like the S3 bucket credentials) could be stored in the ops-dev repo and passed in through the values file. But then we'd have to let other steps in the workflow (that have nothing to do with helm) read this configuration from a renku values file, which might be a bit backwards. So I thought keeping them as separate github secrets and passing them to the helm command separately through another mechanism might be a bit cleaner. 
But if we can make the `additional_values` disappear completely, that's also perfectly fine for me. I just wanted to keep the script free from hardcoded configuration because I'me already getting used to running it manually on my machine. 😃

Reply:
can this be generalized and one can set an environment variable like `TEST_ARTIFACTS_PATH`? Then we can just use this variable everywhere and we don't need to pass in additional options for now.

Reply:
https://github.com/SwissDataScienceCenter/ops-dev.renku.ch/pull/116
It's true that the step where we download the artifact will have to read the values file to fetch the S3 credentials.
Seems a reasonable workaround to keep things general and not duplicate secrets though.

Reply:
I meant we can just pass that env variable to both steps. not beautiful, but it would work.

Reply:
> Can this be generalized and one can set an environment variable like TEST_ARTIFACTS_PATH? Then we can just use this variable everywhere and we don't need to pass in additional options for now.

Not sure I get it - how would you pass this environment variable to helm?

Reply:
the deploy script would set the value

Reply:
e.g.

```
if os.environ.get("TEST_ARTIFACTS_PATH"):
  helm_command += ["--set", "tests.parameters.path=\"$TEST_ARTIFACTS_PATH\""]
```

Reply:
Ah yeah, so we would still handle this as an extra value in the deploy dev renku script, just not provide a generic mechanism for some further additional values. Ok for me. 👍